core/appmodule/module.go

```diff
@@ -83,3 +70,57 @@ type HasEndBlocker interface {
 	// a block.
 	EndBlock(context.Context) error
 }
+
+// HasTxValidation is the extension interface that modules should implement to run
+// custom logic for validating transactions.
+// It was previously known as AnteHandler/Decorator.
+type HasTxValidation interface {
```

Comment:
this interface should be: 

```go
type HasTxValidation[T transaction.Tx]  interface {
     TxValidator(ctx context.Context, t TX) err
}
```

Reply:
And when runtime is wiring it, since runtime expects servertx.Tx should do:

```
txValConcrete, ok := module.(HasTxValidation[servertx.Tx])
if ok { wire to builder normally }
txValInterface, ok := module.(HasTxValidation[transaction.Tx])
if ok {
   wrapper := func(ctx context.Context, tx servertx.Tx) { return txValInterface.TxValidator(ctx context.Context, tx) error)
}
```